docrane/container.py

```diff
@@ -26,20 +30,25 @@ def update_params(self, params):
 
         args:
             container (obj) - Container
-            params (dict) - New params to check
+            etcd_params (dict) - Current params in etcd
 
         Returns: (bool)
             Let us know if they have changed
         """
         has_changed = False
 
-        for param, val in params.iteritems():
+        for param, val in etcd_params.iteritems():
             # Check for new or changed params
             if (param not in self.params.keys() or
                     val != self.params.get(param)):
                 self.set_or_create_param(param, val)
                 has_changed = True
 
+        for param in self.params.keys():
+            if (param not in etcd_params.keys()):
+                self.remove_param(param)
+                has_changed = True
+
```

Comment:
maybe you can get into sets here,

```python
foo = set.symmetrical_difference(set(etcd_params.items()), set(self.params.items()))
# or
foo = set(etcd_params.items()) ^ set(self.params.items())
# then
if len(foo):
    self.params = etcd_params
    LOG.info('params changed %s' % foo)
    return True
```

will puke out a list of key/val pairs that are different between the two, missing or added. If there is a difference, do the fancy logging using the output of symmetrical difference, and update params whole hog. I don't think there's a reason to do it one-per, when we know that `etcd_params` is what we want the container to run with once all is said and done.

and it gets rid of that `has_changed` blight.